client/services/CodeStorageServiceSpec.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the CodeStorageService.
+ */
+
+describe('CodeStorageService', function() {
+  var CodeStorageService;
+  var questionIds = [];
+  var studentSubmittedCodes = [];
+  var i;
+  for (i = 0; i < 5; i++) {
+    questionIds[i] = 'question_' + (i + 1);
+    studentSubmittedCodes[i] = 'code_' + (i + 1);
+  }
+  const language = 'python';
+  const failedLanguage = 'java';
+
+  var generateRandomChars = function(number) {
+    var generatedChars = "";
+    var possible = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+
+    for (i = 0; i < number; i++) {
+      generatedChars += possible.charAt(
+        Math.floor(Math.random() * possible.length));
+    }
+    return generatedChars;
+  };
+
+  var randomQuestionIds = [];
+  var randomQuestionCodes = [];
+
+  const questionIdLenRange = 10;
+  const codeLenRange = 10000;
+  const numOfCases = 5;
+
+  for (i = 0; i < numOfCases; i++) {
+    randomQuestionIds[i] = generateRandomChars(
+      Math.floor(Math.random() * questionIdLenRange));
+    randomQuestionCodes[i] = generateRandomChars(
+      Math.floor(Math.random() * codeLenRange));
+  }
+  
+  beforeEach(module('tie'));
+  beforeEach(inject(function($injector) {
```

Comment:
It might not be a bad idea to generate the random questions and code beforeEach function, rather than only once.